api/rpc/stat/stat.go

```diff
@@ -1,44 +1,239 @@
+	res, err := s.Influx.Query(ql)
+	if err != nil {
+		return nil, err
+	}
+	data := res.Results[0].Series[0].Values[0]
+	entry := IOEntry{
+		Id:     			ID,
+		Name:   			data[0].(string),
+		NumberRead: 	data[1].(int64),
+		NumberWrite:  data[2].(int64),
+		NumberTotal: 	data[3].(int64),
+		SizeRead:			data[4].(int64),
+		SizeWrite:		data[5].(int64),
+		SizeTotal:		data[6].(int64),
+	}
+	return &entry, nil
 }
 
-// ExecuteQuery implements business logic for StatServer interface
-func (s *Stat) ExecuteQuery(ctx context.Context, req *QueryRequest) (*QueryReply, error) {
-	resp, err := s.conn.Query(req.Query)
+//NetQuery Extract net information according to StatRequest
+func (s *Stat) NetQuery(ctx context.Context, req *StatRequest) (*NetReply, error) {
+	idFieldName, nameFieldName := getIDNameFields(req)
+	query := buildQueryString(req, idFieldName, "net")
+	res, err := s.Influx.Query(query)
 	if err != nil {
 		return nil, err
 	}
-	data, err := s.conn.Marshal(resp)
+
+	netReply := NetReply{}
+	list := res.Results[0].Series[0].Values
+	netReply.Entries = make([]*NetEntry, len(list))
+	for i, row := range list {
+		entry, err := s.ressourceNetQuery(ctx, req, idFieldName, nameFieldName, row[1].(string))
+		if err != nil {
+			return nil, err
+		}
+		netReply.Entries[i] = entry
+	}
+	return &netReply, nil
+}
+
+//Extract net information for one ressource according to StatRequest
+func (s *Stat) ressourceNetQuery(ctx context.Context, req *StatRequest, idFieldName string, nameFieldName, ID string) (*NetEntry, error) {
+	var netFields = "rx_bytes, rx_errors, tx_bytes, tx_erros"
+	ql := fmt.Sprintf("SELECT %s, %s WHERE %s='%s' ORDER BY time DESC LIMIT 1", nameFieldName, netFields, idFieldName, ID)
+	res, err := s.Influx.Query(ql)
 	if err != nil {
 		return nil, err
 	}
-	rep := &QueryReply{Response: data}
-	return rep, err
+	data := res.Results[0].Series[0].Values[0]
+	entry := NetEntry{
+		Id:     		ID,
+		Name:   		data[0].(string),
+		RxBytes: 		data[1].(int64),
+		RxErrors:  	data[2].(int64),
+		TxBytes: 		data[3].(int64),
+		TxErrors:		data[4].(int64),
+	}
+	return &entry, nil
 }
+
+//Return specific field name for influx query concidering StatRequest
+func getIDNameFields(req *StatRequest) (string, string) {
+	var idFieldName = "com.docker.swarm.node.id"
+	var nameFieldName = "host"
+	if req.Discriminator == "container" {
+		idFieldName = "container_id"
+		nameFieldName="container_name"
+	} else if req.Discriminator == "service" {
+		idFieldName = "com.docker.swarm.service.id"
+		nameFieldName = "com.docker.swarm.service.name"
+	} else {
+		req.Discriminator = "node"
+	}
+	return idFieldName, nameFieldName
+}
+
+
+//Compute the influx 'sql' query string concidering StatRequest
+func buildQueryString(req *StatRequest, ressourceFieldName string, metric string) string {
+	query := fmt.Sprintf("SELECT distinct %s FROM docker_container_%s WHERE container_id!=''", ressourceFieldName, metric)
```

Comment:
Not sure this is the best approach. This query will return all of the id values from the beginning of time. As the measurement grows this will yield undesirable results. Also, a single query can be used to accomplish the same thing. Check the Observabiilty doc related to "group by and limit 1"